### PR TITLE
Add taxoniq Python library

### PR DESCRIPTION
**Pull request by marieBvr**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/taxoniq/ncbi_refseq_accession_db/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{% set name = "ncbi_refseq_accession_db" %}
```
Comment (hmaarrfk) [inline recipes/taxoniq/ncbi_refseq_accession_db/meta.yaml:1]: These recipes should not be nested.

You also removed the "extras" sections which should list you as a maintainer. Please copy and adapt the section accordingly from the example.

Comment (marieBvr) [reply recipes/taxoniq/ncbi_refseq_accession_db/meta.yaml]: Hi,
Thanks, I will add the recipes separately and add the missing sections.

File: recipes/taxoniq/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  sha256: 19fb5d7f85158dc9e80c636635716ce14bb13671abefc9b785257789353f6982
+
+build:
+  number: 0
+  entry_points:
+    - taxoniq=taxoniq.cli:cli
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+
+requirements:
+  host:
```
Comment (hmaarrfk) [inline recipes/taxoniq/meta.yaml:19]:
```suggestion
  build:
    - {{ compiler('c') }}
    host:
```
seems like you need a c compiler

Comment (hmaarrfk) [reply recipes/taxoniq/meta.yaml:19]: Looking at the setup.py it seems to have some cpp extensions as well

File: recipes/taxoniq/LICENSE [hunk 3]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (hmaarrfk) [inline recipes/taxoniq/LICENSE:1]: Can you bring the fact that this file isn't included up upstream?

They can "fix" this by adding a metadata section:
https://github.com/scikit-image/scikit-image/blob/main/setup.cfg#L10

File: recipes/taxoniq/meta.yaml (hunk 2 above)
Comment (marieBvr) [reply recipes/taxoniq/meta.yaml:22]: Thanks, I was just looking into c compiler.

File: recipes/taxoniq/LICENSE (hunk 3 above)
Comment (marieBvr) [reply recipes/taxoniq/LICENSE:1]: What do you mean by bringing it up?

Comment (hmaarrfk) [reply recipes/taxoniq/LICENSE:1]: Open an issue and let them know that their license file is not included in their tar.gz file from:
https://pypi.org/project/taxoniq/#files

they probably want it to be.

Comment (marieBvr) [reply recipes/taxoniq/LICENSE:1]: Thanks, I opened the issue.

**Comment by marieBvr** [comment]
Hi @conda-forge/help-python,
The recipe has passed all checks and is ready for review.
Thank you.

File: recipes/taxoniq/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+test:
+  imports:
+    - taxoniq
+    - taxoniq.vendored
+  commands:
+    - taxoniq --help
+  requires:
+    - coverage
+    - flake8
+    - wheel
```
Comment (hmaarrfk) [inline recipes/taxoniq/meta.yaml:41]:
```suggestion
    - taxoniq --help
    - pip check
  requires:
    - pip
```

Comment (hmaarrfk) [reply recipes/taxoniq/meta.yaml:41]: coverage and flake8 are development libraries. we don't need to lint code at this stage.

The importance of pip check is explained in https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L58

**Pull request closed without merging**